src/Plugin/resource/DataProvider/DataProviderEntity.php

```diff
@@ -198,29 +199,13 @@ public function view($identifier) {
 
       $value = NULL;
 
-      if (!$this->fieldAccess($resource_field)) {
+      if (!$this->methodAccess($resource_field)) {
         // The field does not apply to the current method.
         continue;
       }
 
-      if ($resource_field->getCallback()) {
-        $value = ResourceManager::executeCallback($resource_field->getCallback(), array($wrapper));
-      }
-      elseif ($resource = $resource_field->getResource()) {
-        // TODO: The resource input data in the field definition has changed.
-        // Now it does not need to be keyed by bundle since you don't even need
-        // an entity to use the resource based field.
-
-        // TODO: Make sure that we don't need to fake a ResourceEntity for this.
-        $resource_data_provider = DataProviderResource::init($this->getRequest(), $resource['name'], array(
-          $resource['major_version'],
-          $resource['minor_version'],
-        ));
-        $value = $resource_data_provider->view($identifier);
-      }
-      else {
-        $value = $this->getValue($wrapper, $resource_field);
-      }
+      $source = new DataSourceEMW($this->getAccount(), $wrapper);
```

Comment:
I'm not sold on the ``DataSource`` name in general, in fact ``DataSource`` is probably a better name then ``DataProvider``. 

Maybe we  ``DataInterpreter`` (not sure about it as-well)?